java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+
+task.setRecurrence(recurrence);
+task.save(dataDir + "SetMonthlyEndAfterDateRecurrence_out.msg", TaskSaveFormat.Msg);
+~~~
+### **Monthly Recurrences: NeverEnd Recurrance Type**
+The following code snippet shows you how to end type can be set by using MapiCalendarRecurrenceEndType.NeverEnd.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+MapiCalendarMonthlyRecurrencePattern recurrence = new MapiCalendarMonthlyRecurrencePattern();
+recurrence.setDay(15);
+recurrence.setPeriod(1);
+recurrence.setPatternType(MapiCalendarRecurrencePatternType.Month);
+recurrence.setEndType(MapiCalendarRecurrenceEndType.NeverEnd);
+recurrence.setWeekStartDay(DayOfWeek.Monday);
+~~~
+## **Working with Yearly Recurrences**
+Aspose.Email supports creation of yearly recurrences using MapiCalendarMonthlyRecurrencePattern. By setting the period property to 12, we can achieve the yearly recurrence pattern. Three different Mapi calendar recurrence end types can be used including EndAfterNOccurrences, EndAfterDate and NeverEnd. This section demonstrates the creation of different yearly recurrence patterns.
+### **Yearly Recurrences: EndAfterNOccurrences Recurrance Type**
+In this type of recurrence, number of recurrences is to be set along with other information as follows:
+
+1. Set start, end and due date.
+1. Create a MapiTask.
+1. Set task state to NotAssigned.
+1. Create the monthly recurrence object by setting the properties like PatternType, Period, WeekStartDay, EndType and OccurenceCount.
+1. Set MapiTask.Recurrence property to this monthly recurrence object to achieve the yearly recurrence.
+1. Save this message on disc.
+
+The following code snippet shows you how to create task with recurrence end type as EndAfterNOccurrence.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+Date startDate = getDate(2015, 7, 1);
+Date dueDate = getDate(2015, 7, 1);
+
+MapiTask task = new MapiTask("This is test task", "Sample Body", startDate, dueDate);
+task.setState(MapiTaskState.NotAssigned);
+
+// Set the Yearly recurrence
+MapiCalendarMonthlyRecurrencePattern recurrence = new MapiCalendarMonthlyRecurrencePattern();
+recurrence.setDay(15);
+recurrence.setPeriod(12);
+recurrence.setPatternType(MapiCalendarRecurrencePatternType.Month);
+recurrence.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+recurrence.setOccurrenceCount(3);
+
+task.setRecurrence(recurrence);
+task.save("Yearly.msg", TaskSaveFormat.Msg);
+~~~
+### **Yearly Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the GetOccurrenceCount() function. This function takes start date , end date and RRULE string. The following code snippet shows you how to creates a recurrence on 15th of each 7th month between start and end by date.
```

Comment:
`*OccurrenceCount* property calculated by the **getOccurrenceCount()** function.`